core/src/main/java/org/elasticsearch/index/query/GeoExecType.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+/** Specifies how a geo query should be run. */
+public enum GeoExecType implements Writeable<GeoExecType> {
+    
+    // Referring back to an example for simplification we discussed earlier today:
+    
+    /*
+     * This enum replaces what was previously represented by string values.
+     * 
+     * However it really only has two valid entries and I don't see this becoming more
+     * (happy to be corrected here). If that is the case - should we try to get rid of
+     * what used to be a string parameter, was turned into an enum by one of my refactoring
+     * steps and simply replace it with a boolean parameter?
+     * */
```

Comment:
I think an enum here is fine as it would be hard to have a boolean parameter where the meaning is properly conveyed for these options.